src/kontrol/utils.py

```diff
@@ -716,3 +716,144 @@ def decode_log_message(token: str, selector: int) -> str | None:
+        // Slot read mask - handle width = 32 case to prevent overflow
+        uint256 mask;
+        if (width == 32) {
+            mask = type(uint256).max;
+        } else {
+            unchecked {
+                mask = (2 ** (8 * width)) - 1;
+            }
+        }
+        // Value right shift
+        uint256 shift = 8 * offset;
+        // Current slot value
+        uint256 slotValue = uint256(vm.load(contractAddress, bytes32(slot)));
+        // Isolate and return data to retrieve
+        return mask & (slotValue >> shift);
+    }
+    function _storeData(address contractAddress, uint256 slot, uint256 offset, uint256 width, uint256 value) internal {
+        // `offset` and `width` must not overflow the slot
+        assert(offset + width <= 32);
+        // and `value` must fit into the designated part
+        assert(width == 32 || value < 2 ** (8 * width));
+        // Slot update mask
+        uint256 maskLeft;
+        unchecked {
+            maskLeft = ~((2 ** (8 * (offset + width))) - 1);
+        }
+        uint256 maskRight = (2 ** (8 * offset)) - 1;
+        uint256 mask = maskLeft | maskRight;
+        uint256 updatedValue = (2 ** (8 * offset)) * value;
+        // Current slot value
+        uint256 slotValue = uint256(vm.load(contractAddress, bytes32(slot)));
+        // Updated slot value
+        slotValue = updatedValue | (mask & slotValue);
+        vm.store(contractAddress, bytes32(slot), bytes32(slotValue));
+    }
+    function _loadMappingData(
+        address contractAddress,
+        uint256 mappingSlot,
+        uint256 key,
+        uint256 subSlot,
+        uint256 offset,
+        uint256 width
+    ) internal view returns (uint256) {
+        bytes32 hashedSlot = keccak256(abi.encodePacked(key, mappingSlot));
+        return _loadData(contractAddress, uint256(hashedSlot) + subSlot, offset, width);
+    }
+    function _storeMappingData(
+        address contractAddress,
+        uint256 mappingSlot,
+        uint256 key,
+        uint256 subSlot,
+        uint256 offset,
+        uint256 width,
+        uint256 value
+    ) internal {
+        bytes32 hashedSlot = keccak256(abi.encodePacked(key, mappingSlot));
+        _storeData(contractAddress, uint256(hashedSlot) + subSlot, offset, width, value);
+    }
+    function _loadUInt256(address contractAddress, uint256 slot) internal view returns (uint256) {
+        return _loadData(contractAddress, slot, 0, 32);
+    }
+    function _loadMappingUInt256(
+        address contractAddress,
+        uint256 mappingSlot,
+        uint256 key,
+        uint256 subSlot
+    ) internal view returns (uint256) {
+        bytes32 hashedSlot = keccak256(abi.encodePacked(key, mappingSlot));
+        return _loadData(contractAddress, uint256(hashedSlot) + subSlot, 0, 32);
+    }
+    function _loadAddress(address contractAddress, uint256 slot) internal view returns (address) {
+        return address(uint160(_loadData(contractAddress, slot, 0, 20)));
+    }
```

Comment:
We could add `_loadBytes32`, and `...Address` and `...Bytes32` versions of `_loadMapping...` and `_storeMapping...` as well. if we want to be complete.